handlers/handlers_test.go

```diff
@@ -0,0 +1,743 @@
+			})
+		})
+
+		Convey("When a valid GET request is made with a redirect URL", func() {
+			handler := FlorenceLoginHandler(ctx, "../static/json/users_test.json", "../templates/user.login.html")
+			request := httptest.NewRequest(http.MethodGet, "/florence/login?redirect=/some/path", nil)
+			responseRecorder := httptest.NewRecorder()
+
+			handler.ServeHTTP(responseRecorder, request)
+
+			Convey("Then it should return 200 OK", func() {
+				So(responseRecorder.Code, ShouldEqual, http.StatusOK)
+			})
+
+			Convey("And the response should contain the redirect URL and user data", func() {
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "/some/path")
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "admin@ons.gov.uk")
+			})
+		})
+
+		Convey("When a valid GET request is made without a redirect URL", func() {
+			handler := FlorenceLoginHandler(ctx, "../static/json/users_test.json", "../templates/user.login.html")
+			request := httptest.NewRequest(http.MethodGet, "/florence/login", nil)
+			responseRecorder := httptest.NewRecorder()
+
+			handler.ServeHTTP(responseRecorder, request)
+
+			Convey("Then it should return 200 OK", func() {
+				So(responseRecorder.Code, ShouldEqual, http.StatusOK)
+			})
+
+			Convey("And the response should contain the default redirect URL", func() {
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "/florence/collections")
+			})
+		})
+
+		Convey("When the users file is missing", func() {
+			Handler := FlorenceLoginHandler(ctx, "../static/json/invalid_users.json", "../templates/user.login.html")
+			request := httptest.NewRequest(http.MethodGet, "/florence/login", nil)
+			responseRecorder := httptest.NewRecorder()
+
+			Handler.ServeHTTP(responseRecorder, request)
+
+			Convey("Then it should return 500 Internal Server Error", func() {
+				So(responseRecorder.Code, ShouldEqual, http.StatusInternalServerError)
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "no such file or directory")
+			})
+		})
+
+		Convey("When the template file is missing", func() {
+			Handler := FlorenceLoginHandler(ctx, "../static/json/users_test.json", "../templates/invalid_template.html")
+			request := httptest.NewRequest(http.MethodGet, "/florence/login", nil)
+			responseRecorder := httptest.NewRecorder()
+
+			Handler.ServeHTTP(responseRecorder, request)
+
+			Convey("Then it should return 500 Internal Server Error", func() {
+				So(responseRecorder.Code, ShouldEqual, http.StatusInternalServerError)
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "no such file or directory")
+			})
+		})
+
+	})
+}
+
+func TestFlorenceLoginHandlerPOST(t *testing.T) {
+	Convey("Given a context, usersFile, privateKeyPath and a FlorenceLoginHandlerPOST", t, func() {
+		ctx := context.Background()
+
+		Convey("When the request method is not POST", func() {
+			handler := FlorenceLoginHandlerPOST(ctx, "../static/json/users_test.json", "../static/keys/private.key")
```

Comment:
Same thing here regarding global variables and `"../static/keys/private.key"`

Reply:
Done